### DISCUSSION TITLE
Connessione HTTPS con ha core

### DISCUSSION THREAD

**Discussion by simone929** [indomus/forum#284] [HUB: Home Assistant]
Ciao a tutti, sto seguendo la giuda "[_Collegarsi da remoto a Home Assistant (CORE) installato su Raspbian di Raspberry Pi_](https://indomus.it/guide/collegarsi-da-remoto-a-home-assistant-installato-su-raspberry-raspbian/)" ma già parto male nel senso che avendo seguito quella per l'installazione di Home Assistant Core, non ho lo user homeassistant per cui il passaggio 
_"Una volta collegati via SSH al Raspberry Pi, cambiare l’utente in uso in “homeassistant”:
sudo su -s /bin/bash homeassistant"_ mi restituisce errore in quanto l'utente non esiste.
Comunque sono andato avanti anteponendo davanti ai vari comandi il sudo. tutto bene fino al punto 
_"Eseguire poi il seguente comando:
./dehydrated -c"_ 
in quanto l'output, nelle ultime righe, mi restituisce questo avviso:

_Unknown hook sync_cert
Failed to restart home-assistant@homeassistant.service: Unit home-assistant@homeassistant.service not found.
ERROR: deploy_cert hook returned with non-zero exit code_

Andando avanti infatti con la procedura, dopo aver aggiunto il blocco https al file di configurazione, verificando la configurazione ricevo questo errore:

_Invalid config for [http]: not a file for dictionary value @ data['http']['ssl_certificate']. Got '/home/homeassistant/dehydrated/certs/xxxxxxxx.duckdns.org/fullchain.pem'
not a file for dictionary value @ data['http']['ssl_key']. Got '/home/homeassistant/dehydrated/certs/xxxxxxxx.duckdns.org/privkey.pem'. (See /config/configuration.yaml, line 27)._

Per ora ho rimosso tutto, mi date cortesemente una mano?
Vi ringrazio anticipatamente.
Simone

[output.txt](https://github.com/indomus/forum/files/6445897/output.txt)

**Comment by snakuzzo** [comment]
Premesso che la presenza dell'utente `homeassistant` non è importante ai fini dell'installazione, se hai seguito la [NOSTRA GUIDA](https://indomus.it/guide/come-installare-e-configurare-home-assistant-su-un-raspberry-pi-gia-in-uso/), la creazione dell'utente è prevista.

L'errore finale è perché probabilmente l'utente con cui viene avviato HA non ha i permessi in lettura sui file dei certificati

**Comment by simone929** [comment]
Ciao snakuzzo, grazie per il riscontro. In realtà però ho utilizzato la procedura di installazione tramite docker e li non è prevista la creazione dell'utenza. Ora ho visto che le cartelle di home assistant sono root:root, provo a ripetere.la procedura o tramite root o faccio un chmod -R pi:pi. Pensi possa funzionare?

**Reply by snakuzzo** [reply]
Se hai installato su docker ti consiglio di seguire la nostra guida per [Caddy](https://indomus.it/guide/come-installare-e-configurare-caddy-con-docker-su-raspbian-di-raspberry-pi/)...lascia perdere dehydrated.

**Reply by simone929** [reply]
Ciao,
provando ad installare caddy ricevo questo errore:


pi@raspitre:~ $ docker run --init -d --name="caddy" -e "TZ=Europe/Rome" -v /home/pi/caddy/Caddyfile:/etc/caddy/Caddyfile -v /home/pi/caddy/data:/data --restart always -p 443:443 -p 80:80 caddy:latest
Unable to find image 'caddy:latest' locally
latest: Pulling from library/caddy
420c7481a3a7: Already exists
b239ad5e6984: Pull complete
8a0460b038be: Pull complete
0905d1ca8c4f: Pull complete
2f02da274f55: Pull complete
Digest: sha256:88fc125055b8c4dafe67a48219fd3a89c0a472aa254d2835b2e298b93493b28e
Status: Downloaded newer image for caddy:latest
680ceb09c8ca810ac06fc57edf5bdb8a4b6c5a5dc58f84f547d4878306570b0a
_docker: Error response from daemon: driver failed programming external connectivity on endpoint caddy (81e5d63ce348235dba9f4825267b824231d9f430a6690295ae3b04aafc0426ba): Error starting userland proxy: listen tcp4 0.0.0.0:80: bind: address already in use._

**Reply by snakuzzo** [reply]
La porta 80 è già in uso da un altro servizio. Hai già un webserver di qualcosa sulla porta 80

**Reply by simone929** [reply]
si esatto, ho il monitoraggio dell'ups...posso cambiarla?

**Reply by snakuzzo** [reply]
la porta 80 ti serve per caddy per la challenge per il rinnovo del certificato. Altrimenti in alternativa puoi NON esporre la porta 80 con Caddy ed esporre solo la 443 (per la challenge) e le altre che ti servono (es. la 8123 di HA).
Se non sai cosa siano le challenge, leggi [QUI](https://indomus.it/guide/come-installare-e-configurare-caddy-con-docker-su-raspbian-di-raspberry-pi/)

**Comment by cutdownrecords** [comment]
ho lo stesso problema ho installato HA corem ho seguito la vostra guida passo passo dopo 3 o 4 prove ho tentato anche con root, ma il risultato non cambia.
+ Creating new directory /home/homeassistant/dehydrated/certs/XXX.duckdns.org ...
Unknown hook this_hookscript_is_broken__dehydrated_is_working_fine__please_ignore_unknown_hooks_in_your_script
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting new certificate order from CA...
 + Received 1 authorizations URLs from the CA
 + Handling authorization for XXX.duckdns.org
 + 1 pending challenge(s)
 + Deploying challenge tokens...
OK
 + Responding to challenge for XXX.duckdns.org authorization...
 + Challenge is valid!
 + Cleaning challenge tokens...
OK
 + Requesting certificate...
 + Checking certificate...
 + Done!
 + Creating fullchain.pem...
**Unknown hook sync_cert**
 + Done!


questi sono i log di quando provo a connettermi

File "/srv/homeassistant/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 334, in data_received
    messages, upgraded, tail = self._request_parser.feed_data(data)
  File "aiohttp/_http_parser.pyx", line 551, in aiohttp._http_parser.HttpParser.feed_data
aiohttp.http_exceptions.BadStatusLine: 400, message="Bad status line 'Invalid method encountered'"
2021-12-14 12:33:36 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 334, in data_received
    messages, upgraded, tail = self._request_parser.feed_data(data)
  File "aiohttp/_http_parser.pyx", line 551, in aiohttp._http_parser.HttpParser.feed_data
aiohttp.http_exceptions.BadStatusLine: 400, message="Bad status line 'Invalid method encountered'"


qualcuno può aiutarmi? Al momento sono impossibilitato a installare su docker in quanto ho un router fastgate e porte 80 22 e 443 sono occupate da loro
